Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/dialogs/IniEditorDialog.java

```diff
@@ -0,0 +1,146 @@
+package org.dolphinemu.dolphinemu.dialogs;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.os.Bundle;
+import androidx.annotation.NonNull;
+import androidx.fragment.app.DialogFragment;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import org.dolphinemu.dolphinemu.R;
+import org.dolphinemu.dolphinemu.model.GameFile;
+import org.dolphinemu.dolphinemu.services.GameFileCacheManager;
+import org.dolphinemu.dolphinemu.utils.DirectoryInitialization;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+
+public class IniEditorDialog extends DialogFragment
+{
+    private static final String ARG_GAME_PATH = "game_path";
+
+    public static IniEditorDialog newInstance(String gamePath)
+    {
+        IniEditorDialog fragment = new IniEditorDialog();
+
+        Bundle arguments = new Bundle();
+        arguments.putString(ARG_GAME_PATH, gamePath);
+        fragment.setArguments(arguments);
+
+        return fragment;
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState)
+    {
+        final GameFile gameFile =
+                GameFileCacheManager.addOrGet(getArguments().getString(ARG_GAME_PATH));
+        final String gameId = gameFile.getGameId();
```

Comment:
If the game ID is all you want, you should pass the game ID to the constructor instead of the path.